src/commands/cmd_docker.nim

```diff
@@ -176,18 +177,18 @@ proc writeNewDockerFileIfNeeded(ctx: DockerInvocation) =
   # should properly be using a temporary file, because it's a place
   # we're generally guaranteed to be able to write.
 
-  let (f, path) = getNewTempFile()
-
-  info("Created temporary Dockerfile at: " & path)
-
+  # add last blank line
   if ctx.inDockerFile.len() != 0 and ctx.inDockerFile[^1] != '\n':
     ctx.inDockerFile &= "\n"
-
   let newcontents = ctx.inDockerFile & ctx.addedInstructions.join("\n")
 
-  trace("New docker file: \n" & newcontents)
-  f.write(newcontents)
-  f.close()
+  let (f, path) = getNewTempFile()
```

Comment:
This should have an analogue `writeNewTempFile()` that does the stuff you do below.

Reply:
good idea. added in https://github.com/crashappsec/nimutils/commit/c5235862c40461068a7e929a9487dc9ab27d2495